File: iOS/AUiKit/AUiKit/Classes/Service/Impl/AUiInvitationServiceImpl.swift
//
//  AUiInvitationServiceImpl.swift
//  AUiKit
//
//  Created by wushengtao on 2023/2/23.
//

import Foundation
import AgoraRtcKit

//邀请Service实现
open class AUiInvitationServiceImpl: NSObject {
    private var respDelegates: NSHashTable<AnyObject> = NSHashTable<AnyObject>.weakObjects()
    private var channelName: String!
    private var rtmManager: AUiRtmManager!
    
    deinit {
        aui_info("deinit AUiInvitationServiceImpl", tag: "AUiInvitationServiceImpl")
    }
    
    init(channelName: String, rtmManager: AUiRtmManager) {
        self.channelName = channelName
        self.rtmManager = rtmManager
        super.init()
        
        aui_info("init AUiInvitationServiceImpl", tag: "AUiInvitationServiceImpl")
    }
}

extension AUiInvitationServiceImpl: AUiInvitationServiceDelegate {
    public func getChannelName() -> String {
        return channelName
    }
    
    public func bindRespDelegate(delegate: AUiInvitationRespDelegate) {
        respDelegates.add(delegate)
    }
    
    public func unbindRespDelegate(delegate: AUiInvitationRespDelegate) {
        respDelegates.remove(delegate)
    }
    
    public func sendInvitation(userId: String, seatIndex: Int?, callback: (Error?) -> ()) {
        
    }
    
    public func acceptInvitation(userId: String, seatIndex: Int?, callback: (Error?) -> ()) {
        
    }
    
    public func rejectInvitation(userId: String, callback: (Error?) -> ()) {
    }
    
    public func cancelInvitation(userId: String, callback: (Error?) -> ()) {
    }
    
    public func sendApply(seatIndex: Int?, callback: (Error?) -> ()) {
        
    }
    
    public func cancelApply(callback: (Error?) -> ()) {
        
    }
    
    public func acceptApply(userId: String, seatIndex: Int?, callback: (Error?) -> ()) {
        
    }
    
    public func rejectApply(userId: String, callback: (Error?) -> ()) {
        
    }
    
}

File: Android/auikit/src/main/java/io/agora/auikit/service/imp/AUiInvitationServiceImpl.kt
package io.agora.auikit.service.imp

import io.agora.auikit.model.AUiRoomContext
import io.agora.auikit.service.IAUiInvitationService
import io.agora.auikit.service.callback.AUiCallback
import io.agora.auikit.service.rtm.AUiRtmManager
import io.agora.auikit.utils.DelegateHelper

class AUiInvitationServiceImpl(
    private val roomContext: AUiRoomContext,
    private val channelName: String,
    private val rtmManager: AUiRtmManager
) : IAUiInvitationService {

    private val delegateHelper = DelegateHelper<IAUiInvitationService.AUiInvitationRespDelegate>()

    override fun bindRespDelegate(delegate: IAUiInvitationService.AUiInvitationRespDelegate?) {
        delegateHelper.bindDelegate(delegate)
    }

    override fun unbindRespDelegate(delegate: IAUiInvitationService.AUiInvitationRespDelegate?) {
        delegateHelper.unBindDelegate(delegate)
    }

    override fun sendInvitation(
        cmd: String?,
        userId: String,
        content: String?,
        callback: AUiCallback?
    ) {
        TODO("Not yet implemented")
    }

    override fun acceptInvitation(id: String, callback: AUiCallback?) {
        TODO("Not yet implemented")
    }

    override fun rejectInvitation(id: String, callback: AUiCallback?) {
        TODO("Not yet implemented")
    }

    override fun cancelInvitation(id: String, callback: AUiCallback?) {
        TODO("Not yet implemented")
    }

    override fun getChannelName() = channelName
}